common/metrics/metricstest/metricstest.go

```diff
@@ -193,10 +244,52 @@ func (s Snapshot) MustGauge(name string, tags ...metrics.Tag) float64 {
 	return v
 }
 
+func (s Snapshot) Histogram(name string, tags ...metrics.Tag) ([]HistogramBucket, error) {
+	labelValues := map[string]string{}
+	for _, tag := range tags {
+		labelValues[tag.Key()] = tag.Value()
+	}
+
+	sample, ok := s.histogramSamples[name]
+	if !ok {
+		return nil, fmt.Errorf("metric %s not found", name)
+	}
+	if sample.metricType != dto.MetricType_HISTOGRAM {
+		return nil, fmt.Errorf("metric %s not a %s type", name, dto.MetricType_HISTOGRAM.String())
+	}
+	if !maps.Equal(sample.labelValues, labelValues) {
+		return nil, fmt.Errorf("metric %s label mismatch, has %v, asked for %v", name, sample.labelValues, labelValues)
+	}
+	return sample.buckets, nil
+}
+
+func (s Snapshot) MustHistogram(name string, tags ...metrics.Tag) []HistogramBucket {
+	v, err := s.Histogram(name, tags...)
+	if err != nil {
+		panic(err)
+	}
+	return v
+}
+
 func (s Snapshot) String() string {
 	var b strings.Builder
```

Comment:
If you look at the impl. for `strings.Builder`, you'll find this:

```go
// WriteString appends the contents of s to b's buffer.
// It returns the length of s and a nil error.
func (b *Builder) WriteString(s string) (int, error) {
	b.copyCheck()
	b.buf = append(b.buf, s...)
	return len(s), nil
}
```

So, this method never returns an error. Because of this, please explicitly ignore these errors e.g. `_, _ = b.WriteString(...)` because we're using an implementation that will never return an error.